pyresample/spherical.py

```diff
@@ -176,9 +253,16 @@ def __rmul__(self, other):
         return self.__mul__(other)
 
     def to_spherical(self):
-        """Convert to Spherical coordinate object."""
-        return SCoordinate(np.arctan2(self.cart[1], self.cart[0]),
-                           np.arcsin(self.cart[2]))
+        """Convert to SCoordinate object."""
+        if self.cart.ndim == 1:
+            lon = np.arctan2(self.cart[1], self.cart[0])
+            lat = np.arcsin(self.cart[2])
+        else:
+            lon = np.arctan2(self.cart[:, 1], self.cart[:, 0])
+            lat = np.arcsin(self.cart[:, 2])
+        # TODO: in future this should point to SPoint or SMultiPoint
+        # - This is not used within the file.
```

Comment:
Can it be done now?

Reply:
[...,1]

Reply:
Would introduce backward incompatibility.

Reply:
Can we put this stuff under a feature flag so there is some pyresample global option (environment variable if we have to) to opt-in to these new features?